Documentation/CRDs/Cluster/ceph-cluster-crd.md

```diff
@@ -478,7 +478,7 @@ See the table in [OSD Configuration Settings](#osd-configuration-settings) to kn
 
 The following storage selection settings are specific to Ceph and do not apply to other backends. All variables are key-value pairs represented as strings.
 
-* `metadataDevice`: Name of a device or lvm to use for the metadata of OSDs on each node.  Performance can be improved by using a low latency device (such as SSD or NVMe) as the metadata device, while other spinning platter (HDD) devices on a node are used to store data. Provisioning will fail if the user specifies a `metadataDevice` but that device is not used as a metadata device by Ceph. Notably, `ceph-volume` will not use a device of the same device class (HDD, SSD, NVMe) as OSD devices for metadata, resulting in this failure.
+* `metadataDevice`: Name of a device, partition or lvm to use for the metadata of OSDs on each node.  Performance can be improved by using a low latency device (such as SSD or NVMe) as the metadata device, while other spinning platter (HDD) devices on a node are used to store data. Provisioning will fail if the user specifies a `metadataDevice` but that device is not used as a metadata device by Ceph. Notably, `ceph-volume` will not use a device of the same device class (HDD, SSD, NVMe) as OSD devices for metadata, resulting in this failure.
```

Comment:
Could you write down the limitation of specifying partition? To minimize the effect of this PR to the existent users, the expected behavior should be as follows.

- If `metadataDevice` is specified in the global OSD configuration or in the node level OSD configuration, the metadata device will be shared between all OSDs on the same node as before. In other words, OSDs will be initialized by `lvm batch`. In this case, we can't use partition.
- If `metadataDevice` is specified in the device local configuration,  we can use partition as metadata device. In other words, OSDs are initialized by `lvm prepare`.

If the current behavior is not as written in above, please fix it.

Reply:
OK, I will update it ASAP.

Reply:
I have added a section of limitations for metadata device, PTAL, thanks. @satoru-takeuchi